src/content-tags-drawer/ContentTagsCollapsible.test.jsx

```diff
@@ -200,6 +239,66 @@ describe('<ContentTagsCollapsible />', () => {
     });
   });
 
+  it('should handle search term change', async () => {
+    jest.useFakeTimers(); // To account for debounce timer
+
+    await act(async () => {
+      const {
+        container,
+        getByRole,
+        getByText,
+        getByDisplayValue,
+      } = render(
+        <ContentTagsCollapsibleComponent
+          contentId={data.contentId}
+          taxonomyAndTagsData={data.taxonomyAndTagsData}
+          editable={data.editable}
+        />,
+      );
+
+      // Expand the Taxonomy to view applied tags and "Add tags" button
+      const expandToggle = container.getElementsByClassName('collapsible-trigger')[0];
+      await act(async () => {
+        fireEvent.click(expandToggle);
+      });
+
+      // Click on "Add tags" button to open dropdown
+      const addTagsButton = getByText(messages.addTagsButtonText.defaultMessage);
+      await act(async () => {
+        fireEvent.click(addTagsButton);
+      });
+
+      // Get the search field
+      const searchField = getByRole('searchbox');
+
+      const searchTerm = 'memo';
+
+      // Trigger a change in the search field
+      await userEvent.type(searchField, searchTerm);
+
+      // Fast-forward time by 500 milliseconds (for the debounce delay)
+      act(() => {
+        jest.advanceTimersByTime(500);
+      });
+
+      await waitFor(() => {
+        // Check that the search term has been set
+        expect(searchField.value).toBe(searchTerm);
+        expect(getByDisplayValue(searchTerm)).toBeInTheDocument();
+      });
+
+      // Clear search
+      await userEvent.clear(searchField);
+
+      await waitFor(() => {
+        // Check that the search term has been set
+        expect(searchField.value).toBe('');
+      });
+
+      jest.useRealTimers(); // Restore real timers after the test
```

Comment:
I think the timer mocks should be set up in a `beforeAll` or `beforeEach` and `afterAll` or `afterEach` setup and teardown sections, otherwise if this test fails the timers won't be reset AFAIK.

Reply:
Updated here: https://github.com/openedx/frontend-app-course-authoring/pull/737/commits/9789faf9585d122fbcb76b241d9f79ff917a24da